### PR TITLE
UCPKN-2281: Implement Day sub-entity.

### PR DESCRIPTION
**Pull request by kp77**
#2 needs to be merged first.

### CODE REVIEW

**Review by dxvargas — DISMISSED** [review]

File: config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml
Comment (brummbar) [inline config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml:1]: Why don't use use the label base field?

File: src/EventSubscriber/DayGenerateLabelSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+   */
+  protected function generateLabel(ContentEntityInterface $entity): ?string {
+    if ($entity->bundle() === 'oe_default'
+      && $entity->hasField('oe_day_title')
+      && $entity->hasField('oe_day_date')
+    ) {
+      // Display the day title as label if set, otherwise the date for Default
+      // day type.
+      return $entity->get('oe_day_title')->isEmpty()
+        ? $entity->oe_day_date->value
```
Comment (brummbar) [inline src/EventSubscriber/DayGenerateLabelSubscriber.php]: Let's use the `->get('field_name')->` format.

File: tests/src/FunctionalJavascript/DayEntityTest.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    $page->fillField('field_agenda[form][0][oe_agenda_days][form][1][oe_day_date][0][value][date]', '08-22-2023');
+    $page->pressButton('Create day');
+    $assert_session->assertWaitOnAjaxRequest();
+    $assert_session->elementTextContains('css', '.ief-entity-table tbody tr:nth-child(2) td.inline-entity-form-oe_agenda_day-label', '2023-08-22');
+    // Test removing second day.
+    $page->pressButton('ief-field_agenda-form-0-oe_agenda_days-form-entity-remove-1');
+    $assert_session->assertWaitOnAjaxRequest();
+    $page->pressButton('ief-remove-confirm-field_agenda-form-0-oe_agenda_days-form-1');
+    $assert_session->assertWaitOnAjaxRequest();
+    $assert_session->pageTextNotContains('2023-08-22');
```
Comment (brummbar) [inline tests/src/FunctionalJavascript/DayEntityTest.php:71]: Let's also add a check on the rendered values in the frontend.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml
Comment (kp77) [reply config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml:1]: Hi @brummbar,

Currently, the Day doesn't have a label base field, we'd need to define one for this. I was also thinking about this when implementing, but I decided that the tech specification lists an optional title bundle field for a reason, and we don't need a title/label base field for the day (as giving a title to a day is optional). What do you think?

File: src/EventSubscriber/DayGenerateLabelSubscriber.php (hunk 1 above)
Comment (kp77) [reply src/EventSubscriber/DayGenerateLabelSubscriber.php]: changed it

File: tests/src/FunctionalJavascript/DayEntityTest.php (hunk 2 above)
Comment (kp77) [reply tests/src/FunctionalJavascript/DayEntityTest.php:71]: added some value assertions on the node page

File: config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml
Comment (brummbar) [reply config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml:1]: I think we should put the label into the entity definition, and leave it as optional field.

Comment (kp77) [reply config/install/field.field.oe_agenda_day.oe_default.oe_day_title.yml:1]: I replaced the title with a base field in [ad224c0](https://github.com/openeuropa/oe_agenda/pull/3/commits/ad224c0bb14a059e4c9de6a0097a4961cf408d21) (I called it Title, as that's what the field is called in the specs, it's similar to node title, only not required).

**Review by brummbar — APPROVED** [review]

**Review by dxvargas — APPROVED** [review]

**Pull request merged**